File: components/ide-ui-security/src/main/resources/META-INF/dirigible/ide-security/views/roles.js
/*
 * Copyright (c) 2022 SAP SE or an SAP affiliate company and Eclipse Dirigible contributors
 *
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v2.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v20.html
 *
 * SPDX-FileCopyrightText: 2022 SAP SE or an SAP affiliate company and Eclipse Dirigible contributors
 * SPDX-License-Identifier: EPL-2.0
 */
const rolesView = angular.module('roles', ['ideUI', 'ideView']);

rolesView.config(["messageHubProvider", function (messageHubProvider) {
	messageHubProvider.eventIdPrefix = 'roles-view';
}]);

rolesView.controller('RolesController', ['$scope', '$http', function ($scope, $http) {

	$http.get('/services/ops/security/roles').then(function (response) {
		$scope.list = response.data;
	});

}]);

File: components/core/core-base/src/main/java/org/eclipse/dirigible/components/base/http/roles/Roles.java
/*
 * Copyright (c) 2023 SAP SE or an SAP affiliate company and Eclipse Dirigible contributors
 *
 * All rights reserved. This program and the accompanying materials are made available under the
 * terms of the Eclipse Public License v2.0 which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v20.html
 *
 * SPDX-FileCopyrightText: 2023 SAP SE or an SAP affiliate company and Eclipse Dirigible
 * contributors SPDX-License-Identifier: EPL-2.0
 */
package org.eclipse.dirigible.components.base.http.roles;

/**
 * The Enum Roles.
 */
public enum Roles {

    /** The administrator. */
    ADMINISTRATOR("ROLE_ADMINISTRATOR"),
    /** The developer. */
    DEVELOPER("ROLE_DEVELOPER"),
    /** The operator. */
    OPERATOR("ROLE_OPERATOR");

    /** The role name. */
    private final String roleName;

    /**
     * Instantiates a new roles.
     *
     * @param roleName the role name
     */
    Roles(String roleName) {
        this.roleName = roleName;
    }

    /**
     * Gets the role name.
     *
     * @return the role name
     */
    public String getRoleName() {
        return roleName;
    }

}
